File: android/worker/app/src/main/java/com/workforcex/worker/ui/worker/BrowseJobsAdapter.java
package com.workforcex.worker.ui.worker;

import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.workforcex.worker.R;
import com.workforcex.worker.api.JobBrowseItem;
import java.util.List;
import java.util.Set;

public class BrowseJobsAdapter extends RecyclerView.Adapter<BrowseJobsAdapter.JobViewHolder> {

    public interface OnApplyClick { void onApply(JobBrowseItem job, int position); }

    private List<JobBrowseItem> jobs;
    private final OnApplyClick onApplyClick;
    private final Set<String> appliedJobIds;

    public BrowseJobsAdapter(List<JobBrowseItem> jobs, OnApplyClick onApplyClick, Set<String> appliedJobIds) {
        this.jobs = jobs;
        this.onApplyClick = onApplyClick;
        this.appliedJobIds = appliedJobIds;
    }

    public void update(List<JobBrowseItem> newJobs) {
        this.jobs = newJobs;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public JobViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_job_browse, parent, false);
        return new JobViewHolder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull JobViewHolder h, int position) {
        JobBrowseItem job = jobs.get(position);
        h.tvTitle.setText(job.title);
        h.tvCompany.setText(job.companyName != null ? job.companyName : "");
        h.tvLocation.setText("📍 " + (job.location != null ? job.location : "Not specified"));
        h.tvSkills.setText("Skills: " + (job.skillsRequired != null ? job.skillsRequired : "Any"));

        if (job.salaryMin != null && job.salaryMax != null) {
            h.tvSalary.setText("₹" + job.salaryMin.intValue() + " – ₹" + job.salaryMax.intValue() + "/month");
        } else if (job.salaryMin != null) {
            h.tvSalary.setText("₹" + job.salaryMin.intValue() + "+/month");
        } else {
            h.tvSalary.setText("");
        }

        h.tvOpenings.setText(job.openPositions != null && job.openPositions > 0
                ? job.openPositions + " opening" + (job.openPositions > 1 ? "s" : "") : "");

        boolean isApplied = appliedJobIds.contains(job.id) || job.applied;
        if (isApplied) {
            h.btnApply.setEnabled(false);
            h.btnApply.setText("Applied");
        } else {
            h.btnApply.setEnabled(true);
            h.btnApply.setText("Apply Now");
        }

        h.btnApply.setOnClickListener(v -> onApplyClick.onApply(job, h.getAdapterPosition()));

        h.itemView.setOnClickListener(v -> {
            Intent intent = new Intent(v.getContext(), JobDetailsActivity.class);
            intent.putExtra("job", job);
            intent.putExtra("isApplied", isApplied);
            v.getContext().startActivity(intent);
        });
    }

    @Override
    public int getItemCount() { return jobs.size(); }

    static class JobViewHolder extends RecyclerView.ViewHolder {
        TextView tvTitle, tvCompany, tvLocation, tvSkills, tvSalary, tvOpenings;
        Button btnApply;

        JobViewHolder(View v) {
            super(v);
            tvTitle    = v.findViewById(R.id.tvTitle);
            tvCompany  = v.findViewById(R.id.tvCompany);
            tvLocation = v.findViewById(R.id.tvLocation);
            tvSkills   = v.findViewById(R.id.tvSkills);
            tvSalary   = v.findViewById(R.id.tvSalary);
            tvOpenings = v.findViewById(R.id.tvOpenings);
            btnApply   = v.findViewById(R.id.btnApply);
        }
    }
}

File: android/worker/app/src/main/java/com/workforcex/worker/ui/worker/BrowseJobsAdapter.kt
package com.workforcex.worker.ui.worker

import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.workforcex.worker.R
import com.workforcex.shared.models.JobBrowseItem

class BrowseJobsAdapter(
    private var jobs: List<JobBrowseItem>,
    private val onApplyClick: (JobBrowseItem, Int) -> Unit,
    private val appliedJobIds: Set<String>
) : RecyclerView.Adapter<BrowseJobsAdapter.JobViewHolder>() {

    fun update(newJobs: List<JobBrowseItem>) {
        this.jobs = newJobs
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): JobViewHolder {
        val v = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_job_browse, parent, false)
        return JobViewHolder(v)
    }

    override fun onBindViewHolder(h: JobViewHolder, position: Int) {
        val job = jobs[position]
        h.tvTitle.text = job.title
        h.tvCompany.text = job.companyName ?: ""
        h.tvLocation.text = "📍 " + (job.location ?: "Not specified")
        h.tvSkills.text = "Skills: " + (job.skillsRequired ?: "Any")

        if (job.salaryMin != null && job.salaryMax != null) {
            h.tvSalary.text = "₹" + job.salaryMin.toInt() + " – ₹" + job.salaryMax.toInt() + "/month"
        } else if (job.salaryMin != null) {
            h.tvSalary.text = "₹" + job.salaryMin.toInt() + "+/month"
        } else {
            h.tvSalary.text = ""
        }

        h.tvOpenings.text = if (job.openPositions != null && job.openPositions > 0)
            job.openPositions.toString() + " opening" + if (job.openPositions > 1) "s" else ""
        else ""

        val isApplied = appliedJobIds.contains(job.id) || job.applied
        if (isApplied) {
            h.btnApply.isEnabled = false
            h.btnApply.text = "Applied"
        } else {
            h.btnApply.isEnabled = true
            h.btnApply.text = "Apply Now"
        }

        h.btnApply.setOnClickListener { onApplyClick(job, h.adapterPosition) }

        h.itemView.setOnClickListener { v ->
            val intent = Intent(v.context, JobDetailsActivity::class.java)
            intent.putExtra("job", job)
            intent.putExtra("isApplied", isApplied)
            v.context.startActivity(intent)
        }
    }

    override fun getItemCount(): Int = jobs.size

    class JobViewHolder(v: View) : RecyclerView.ViewHolder(v) {
        val tvTitle: TextView = v.findViewById(R.id.tvTitle)
        val tvCompany: TextView = v.findViewById(R.id.tvCompany)
        val tvLocation: TextView = v.findViewById(R.id.tvLocation)
        val tvSkills: TextView = v.findViewById(R.id.tvSkills)
        val tvSalary: TextView = v.findViewById(R.id.tvSalary)
        val tvOpenings: TextView = v.findViewById(R.id.tvOpenings)
        val btnApply: Button = v.findViewById(R.id.btnApply)
    }
}